keras_cv/models/object_detection/yolox/layers/yolox_pafpn.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from tensorflow import keras
+
+from keras_cv.models.__internal__.darknet_utils import CrossStagePartial
+from keras_cv.models.__internal__.darknet_utils import DarknetConvBlock
+from keras_cv.models.__internal__.darknet_utils import DarknetConvBlockDepthwise
+
+
+class YoloXPAFPN(keras.layers.Layer):
+    """The YoloX PAFPN.
+
+    YoloX PAFPN is an FPN layer used in YoloX models. The YoloX PAFPN is based on the
+    feature pyramid module used in Path Aggregation networks (PANet).
+
+    Arguments:
+        depth_multiplier: A float value used to calculate the base depth of the model
+            this changes based on the detection model being used. Defaults to 1.0.
+        width_multiplier: A float value used to calculate the base width of the model
+            this changes based on the detection model being used. Defaults to 1.0.
+        in_channels: A list representing the number of filters in the FPN output.
+            The length of the list will be same as the number of outputs. Defaults to
+            [256, 512, 1024].
+        use_depthwise: a boolean value used to decide whether a depthwise conv block
+            should be used over a regular darknet block. Defaults to False.
+        activation: the activation applied after the BatchNorm layer. One of "silu",
+            "relu" or "leaky_relu". Defaults to "silu".
+    """
+
+    def __init__(
+        self,
+        depth_multiplier=1.0,
+        width_multiplier=1.0,
+        in_channels=[256, 512, 1024],
```

Comment:
you can't have a list as a default parameter because lists are mutable:

```
def x(y=[]):
    if len(y) > 1:
        raise ValueError(y=y)
    y.append(1)

x() # passes
x() # ValueError(y=[1])
```